src/classes/GS_ApplicationStatusController.cls

```diff
@@ -0,0 +1,215 @@
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+ * @author Salesforce.org
+ * @description Controller for getting the org status(is trial? when it expires?, did customer applied for the licenses?).
+ */
+public with sharing class GS_ApplicationStatusController {
+    private static final String APPLICATION_DATE = 'P10_Application_Date';
+    private static final String SANDBOX_PARAMETER = 'Sandbox';
+    private static final String APPLICATION_PARAMETER = 'Application';
+    private static final String TRIAL_EXPIRATION_PARAMETER = 'TrialExpiration';
+    private static final Date DEFAULT_NON_APPLICATED_DATE = Date.parse('01/01/1970');
+    private static final String STATIC_RESOURCE_NAME = 'gsApplicationConfiguration';
+    
+    private Boolean overrideValuesForTesting;
+
+    /**
+     * @description Property to get the testing configuration from Static Resource.
+     * @return Map<String,Object> with the parsed test configuration loaded from the Static Resource.
+     */
+    @TestVisible
+    private static Map<String, Object> testingConfiguration {
+        get {
+            if (testingConfiguration != null) {
+                return testingConfiguration;
+            } else {
+                testingConfiguration = new GS_ApplicationStatusController().getTestingConfiguration();
+                return testingConfiguration;
+            }
+        }
+        set {
+            testingConfiguration = value;
+        }
+    }
+
+    /**
+     * @description Read-only property with the Org queried with the fields: IsSandbox and TrialExpirationDate.
```

Comment:
Separation of Concerns.
Overall, I think all of the organization state specific methods (isActiveOrg, isSandbox, getExpirationDate, getOrgInformation, etc.) belong in a different class, such as `UTIL_OrgState`. Unit tests would be in a related `UTIL_OrgState_TEST` class. This class would then contain only `@auraEnabled` methods that delegate to the `UTIL_OrgState` class, plus other private static methods that are very specific to the controller operation.